### PR TITLE
Multi item template

### PR DESCRIPTION
**Pull request by lewismarshall**
Will allow multiple resources to be created from a single template file.

### CODE REVIEW

**Comment by timgent** [comment]
I think some more docs may be helpful, perhaps with an example

**Comment by lewismarshall** [comment]
@timgent You read my mind... about to update the PR with a full schema doc detailing all features... and the travis fixes :)

**Comment by timgent** [comment]
@lewismarshall Sounds good :)

**Comment by lewismarshall** [comment]
There complete new doc's for the Schema - be good to have them read through for a LGTM @allandegnan @tim maybe?

The new code can be tested (quay.io/ukhomeofficedigital/kb8or:v0.4.0-rc6) and has been tested:
- regression deploys of checking service to dev
- testing of the new templating features for elasticsearch in AWS. Be good to get a LGTM from anyone time permitting @jon-shanks @vaijab @PurpleBooth

**Comment by jon-shanks** [comment]
LGTM

File: docs/schema.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+FileIncludePaths:
+  - ../tmp/kb8or/ci_secrets_certs.yaml
+  - ../tmp/kb8or/ci_secrets.yaml
+```
+
+### Fn::OptionalHashItem
+
+#### Scope Environment file, a_kubernets_resource.yaml
+
+This allows an item to be removed from a
```
Comment (timgent) [inline docs/schema.md]: from a.....?

**Comment by timgent** [comment]
LGTM. The new docs help a lot. Next time I use it I will try to raise some some enhancements about things that could maybe be simplified - there are a tonne of options at the minute and I reckon some could be rationalised.

**Comment by lewismarshall** [comment]
@timgent Totally agree about simplification, sensible defaults may allow for implicit settings to simplify everything.

**Comment by lewismarshall** [comment]
PS merged and pushed @ quay.io/ukhomeofficedigital/kb8or:v0.4.0

**Pull request merged**